crates/paths/src/paths.rs

```diff
@@ -56,7 +56,13 @@ pub fn set_custom_data_dir(dir: &str) -> &'static PathBuf {
         panic!("set_custom_data_dir called after data_dir or config_dir was initialized");
     }
     CUSTOM_DATA_DIR.get_or_init(|| {
-        let path = PathBuf::from(dir);
+        let mut path = PathBuf::from(dir);
+        if path.is_relative() {
+            let abs_path = path
+                .canonicalize()
+                .expect("failed to get custom data dir from relative path");
```

Comment:
There's quite a rustdoc above, let's update that with the new behavior details:

* when the panic is invoked
* noticing that we do a blocking FS operation